packages/fields/src/fields/slug/index.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * WordPress dependencies
+ */
+import type { Field } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+import type { BasePost } from '../../types';
+import { __ } from '@wordpress/i18n';
+import SlugEdit from './slug-edit';
+import SlugView from './slug-view';
+
+const slugField: Field< BasePost > = {
+	id: 'slug',
```

Comment:
What `type` would this field be? I feel we need to make progress on making `type` obligatory and every new field we introduce should have a type.

Reply:
After further consideration, the type should always represent a generic type, not strictly tied to WordPress. In this case, it should be `link`. In the future, we might have:
- A generic field for the `link` type.
- Other fields that work with the `link` type but have additional functionalities that depend on specific settings like this one.


Would it make sense for each field to have a `name`?

Reply:
Based on [this discussion](https://github.com/WordPress/gutenberg/pull/64496#discussion_r1810262284), I updated the type with 4efda741841b3324cf6397baa2e7a0f152ce758f